src/modules/deal-folder/deal-folder.service.ts

```diff
@@ -0,0 +1,182 @@
+import { BadRequestException, Inject, Injectable } from '@nestjs/common';
+import { ConfigType } from '@nestjs/config';
+import SharepointConfig from '@ukef/config/sharepoint.config';
+import { CASE_LIBRARY, DOCUMENT_X0020_STATUS, ENUMS, MAX_FILE_SIZE_BYTES } from '@ukef/constants';
+import { DocumentTypeEnum } from '@ukef/constants/enums/document-type';
+import { SharepointResourceTypeEnum } from '@ukef/constants/enums/sharepoint-resource-type';
+import { DocumentTypeMapper } from '@ukef/modules/deal-folder/document-type-mapper';
+import { DtfsStorageFileService } from '@ukef/modules/dtfs-storage/dtfs-storage-file.service';
+import GraphService from '@ukef/modules/graph/graph.service';
+
+import { UploadFileInDealFolderResponseDto } from './dto/upload-file-in-deal-folder-response.dto';
+type RequiredConfigKeys = 'baseUrl' | 'ukefSharepointName' | 'estoreDocumentTypeIdFieldName';
+
+@Injectable()
+export class DealFolderService {
+  constructor(
+    @Inject(SharepointConfig.KEY)
+    private readonly config: Pick<ConfigType<typeof SharepointConfig>, RequiredConfigKeys>,
+    private readonly documentTypeMapper: DocumentTypeMapper,
+    private readonly dtfsStorageFileService: DtfsStorageFileService,
+    private readonly graphService: GraphService,
+  ) {}
+
+  async uploadFileInDealFolder(
+    fileName: string,
+    fileLocationPath: string,
+    dealId: string,
+    buyerName: string,
+    ukefSiteId: string,
+    documentType: DocumentTypeEnum,
+  ): Promise<UploadFileInDealFolderResponseDto> {
+    const fileSizeInBytes = await this.checkFileIsNotTooLargeAndReturnSizeInBytes(fileName, fileLocationPath);
+
+    const file = await this.downloadFileFromDtfs(fileName, fileLocationPath);
+
+    await this.uploadFileToSharepoint(file, fileSizeInBytes, fileName, dealId, buyerName, ukefSiteId);
+
+    const fileLeafRef = await this.updateFileInformationInSharepoint(fileName, dealId, buyerName, ukefSiteId, documentType);
+
+    return {
+      fileUpload: fileLeafRef,
+    };
+  }
+
+  private async checkFileIsNotTooLargeAndReturnSizeInBytes(fileName: string, fileLocationPath: string): Promise<number> {
+    const fileSizeInBytes = await this.dtfsStorageFileService.getFileProperties(fileName, fileLocationPath).then((response) => response.contentLength);
+    if (fileSizeInBytes > MAX_FILE_SIZE_BYTES) {
```

Comment:
I think `MAX_FILE_SIZE_BYTES` has been renamed to `DTFS_MAX_FILE_SIZE_BYTES` on main, so watch out for that when you rebase